Initial unix domain socket support

paul-guo- opened the pull request:
I will add test cases if needed.

davecramer commented on src/common/comm_server.c:
```diff
@@ -51,6 +54,62 @@ int start_listener() {
 }
 
 /*
+ * Functoin binds the socket and starts listening on it: unix domain socket.
```
s/b Function

davecramer commented on src/common/comm_server.c:
```diff
@@ -51,6 +54,62 @@ int start_listener() {
+}
+
+/*
  * Fuction waits for the socket to accept connection for finite amount of time
```
Different spelling error for Fuction

davecramer commented on src/plc_configuration.c:
```diff
@@ -378,20 +392,39 @@ char *get_sharing_options(plcContainerConf *conf) {
         }
 
-        res = palloc(totallen + 2 * conf->nSharedDirs);
+		if (!conf->isNetworkConnection) {
```
indention ??

paul-guo- commented:
@davecramer typo were fixed. Thanks.

For indentation, I think we should follow the pg convention which seems to be using tab (with 4 space wideth) for indentation. Later we might replace all other space indentation so during development we will use tab for new code change.

paul-guo- commented on src/common/comm_connectivity.c:
```diff
@@ -370,14 +376,67 @@ plcConn *plcConnect(int port) {
+    return result;
+}
+
+
```
Done. Thanks.

paul-guo- commented:
@stanlyxiang 

I changed the code so there is no need to fix R code. Also memset the conf struct so use_network has an explicit value. Fixed another small bug additionally.

Thanks for reviewing.

paul-guo- commented:
I'll merge them at first. Currently pg_regress runs with this setting by default. There will have some extra security related things need to do next.
